add tags for inplace view ops in native_functions.yaml

bdhirsh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66101 Test functionalization pass in python
* #64432 Add a Functionalization pass in core
* **#65412 add tags for inplace view ops in native_functions.yaml**
* #64430 add slice/select/diagonal_scatter variants as primitive ops
* #63094 move some codegen utilities into utils.py

Differential Revision: [D31942094](https://our.internmc.facebook.com/intern/diff/D31942094)

bdhirsh committed:
add tags for inplace view ops in native_functions.yaml

[ghstack-poisoned]

ezyang reviewed (approved):
Will probably end up rewriting this with the full version of the feature but this is short and sweet

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"




[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"




[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"




[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"




[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"




[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"




[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"


Differential Revision: [D31942094](https://our.internmc.facebook.com/intern/diff/D31942094)

[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"


Differential Revision: [D31942094](https://our.internmc.facebook.com/intern/diff/D31942094)

[ghstack-poisoned]

bdhirsh committed:
Update on "add tags for inplace view ops in native_functions.yaml"


Differential Revision: [D31942094](https://our.internmc.facebook.com/intern/diff/D31942094)

[ghstack-poisoned]

kit1980 commented on tools/codegen/gen.py:
```diff
@@ -108,15 +109,28 @@ def parse_native_yaml(path: str) -> ParsedYaml:
                 f"Consider adding 'structured=True' to the delegated operator"
+        if f.tag is not None and f.tag is Tag.inplace_view:
+            base_name = f.func.name.name
+            overload_name = f.func.name.overload_name
```
`overload_name` never used?
